File: mosbysample/src/main/java/wangfeixixi/github/com/mosbysample/f/CFragment.java
package wangfeixixi.github.com.mosbysample.f;

import android.support.annotation.NonNull;
import android.util.Log;

import wangfeixixi.github.com.mosbysample.base.BaseF;

/**
 * Created by xuany on 2018/5/9.
 */

public class CFragment extends BaseF<CView, CPresent> implements CView {
    private String TAG = "CFragment";

    @Override
    protected void onUserVisible(boolean isVisible) {

        Log.d(TAG, "onUserVisible" + isVisible);
    }

    @Override
    protected void initView() {

        tv_name.setText("CFragment");
    }

    @Override
    protected void initData(boolean firstLoad, boolean isVisibleToUser) {

        Log.d(TAG, "firstLoad" + firstLoad + "isVisibleToUser" + isVisibleToUser);
    }


    @NonNull
    @Override
    public CPresent createPresenter() {
        return new CPresent();
    }
}

File: mosbysample/src/main/java/wangfeixixi/github/com/mosbysample/f/CFragment.kt
package wangfeixixi.github.com.mosbysample.f

import android.util.Log
import kotlinx.android.synthetic.main.a_fragment.*
import wangfeixixi.github.com.mosbysample.R

import wangfeixixi.github.com.mosbysample.base.BaseF


class CFragment : BaseF<CView, CPresent>(), CView {
    override fun initContentRes(): Int = R.layout.a_fragment

    private val TAG = "CFragment"

    protected fun onUserVisible(isVisible: Boolean) {

        Log.d(TAG, "onUserVisible$isVisible")
    }

    override fun initView() {
        tv_name.text = TAG

    }

    override fun initData(firstLoad: Boolean, isVisibleToUser: Boolean) {

        Log.d(TAG, "firstLoad" + firstLoad + "isVisibleToUser" + isVisibleToUser)
    }


    override fun createPresenter(): CPresent {
        return CPresent()
    }
}
